include/xrpl/protocol/TER.h

```diff
@@ -139,6 +139,8 @@ enum TEMcodes : TERUnderlyingType {
 
     temARRAY_EMPTY,
     temARRAY_TOO_LARGE,
+    temBAD_ASSET_AMOUNT,
```

Comment:
There's a temBAD_AMOUNT already defined for non-positive amount errors.

Reply:
That's what I mean, don't need to define new ones.

Reply:
Then the request's error message for this one will be inaccurate: `Can only send positive amounts.`
But I intended to return error message as `Malformed: Amount does not match Asset.`

Reply:
could just use `temMALFORMED` then.

Reply:
But I'm fine if you feel strongly about using new codes.

Reply:
Why not change the error message? IIRC that's not considered a breaking change.

Reply:
> Why not change the error message? IIRC that's not considered a breaking change.

the temBAD_AMOUNT is used in other places to check positive amounts. So here I need a new error type. That's why I created a new error type temBAD_ASSET_AMOUNT. The error message is bind to each error type.

Reply:
It's also thrown in the escrow code if the asset isn't XRP.

Reply:
Why not just change the error message, so that it makes sense both in Escrow and here, instead of needing a new error code? I believe changing the error message isn't considered a breaking change/needing an amendment.

Reply:
I can change the message for `temBAD_AMOUNT` to "bad amount provided". What do you think?

Reply:
`Malformed: Bad amount.` would keep it in line with the others.

Reply:
updated in the recent commit